src/scaffold/mod.rs

```diff
@@ -110,10 +110,28 @@ impl<'s> Scaffolder<'s> {
 
     /// Generates Solidity code from a `.tree` file.
     pub fn scaffold(&self, text: &str) -> crate::error::Result<String> {
-        let ast = syntax::parse(text)?;
+        let combiner = hir::combiner::Combiner::new();
+        let trees = combiner.split(text);
+
         let mut discoverer = modifiers::ModifierDiscoverer::new();
-        let modifiers = discoverer.discover(&ast);
-        let hir = hir::translator::Translator::new().translate(&ast, modifiers);
+        let translator = hir::translator::Translator::new();
+
+        let mut errors = Vec::new();
+        let asts: Vec<syntax::ast::Ast> = trees
+            .iter()
+            .filter_map(|&tree| syntax::parse(tree).map_err(|error| errors.push(error)).ok()) // @follow-up - how do we combine and report errors?
+            .collect();
+
+        let hirs: Vec<hir::Hir> = asts
+            .iter()
+            .map(|ast| {
+                let modifiers = discoverer.discover(ast);
+                translator.translate(ast, modifiers)
+            })
+            .collect();
+
+        combiner.verify(&asts).unwrap();
```

Comment:
We need to be very careful with unwraps, because they are very frustrating for users. At the very least, this should be an `expect`. However, in this particular case, we should forward the error to the callee so that it decides how to handle it.